Source/Core/Core/HW/DSP.cpp

```diff
@@ -472,140 +490,156 @@ void UpdateAudioDMA()
   }
 }
 
-static void Do_ARAM_DMA()
+/* Depending on the size ARAM is configured as, the mapping to the underlying physical ARAM can
+ * change. These mappings have been confirmed on hardware.*/
+static std::optional<u32> ARAM_02MB_to_16MB(u32 address)
 {
-  s_dspState.DMAState = 1;
+  address &= 0x3ffffe0;
```

Comment:
Would it make sense to name (ie. as `constexpr`) this mask? Same for the ones below, since they repeat.